mlir/lib/Dialect/Vector/Transforms/VectorLinearize.cpp

```diff
@@ -145,3 +378,24 @@ void mlir::vector::populateVectorLinearizeTypeConversionsAndLegality(
   patterns.add<LinearizeConstant, LinearizeVectorizable>(
       typeConverter, patterns.getContext(), targetBitWidth);
 }
+
+void mlir::vector::populateVectorLinearizeToShuffleRewritePatterns(
+    TypeConverter &typeConverter, RewritePatternSet &patterns,
+    ConversionTarget &target, unsigned int targetBitWidth) {
+  target.markUnknownOpDynamicallyLegal(
```

Comment:
I think you can just do `target.addDynamicallyLegalOp<vector::ShuffleOp>(...)`

Reply:
fixed.